applications/vanilla/models/class.discussionmodel.php

```diff
@@ -2634,32 +2633,10 @@ public function updateDiscussionCount($categoryID, $discussion = false) {
             $this->Database->query($sql, $params, 'DiscussionModel_UpdateDiscussionCount');
 
         } elseif (is_numeric($categoryID)) {
-            $this->SQL
-                ->select('d.DiscussionID', 'count', 'CountDiscussions')
-                ->select('d.CountComments', 'sum', 'CountComments')
-                ->from('Discussion d')
-                ->where('d.CategoryID', $categoryID);
-
-            $data = $this->SQL->get()->firstRow();
-            $countDiscussions = (int)getValue('CountDiscussions', $data, 0);
-            $countComments = (int)getValue('CountComments', $data, 0);
-
-            $cacheAmendment = [
-                'CountDiscussions' => $countDiscussions,
-                'CountComments' => $countComments
-            ];
-
-            if ($discussionID) {
-                $cacheAmendment = array_merge($cacheAmendment, [
-                    'LastDiscussionID' => $discussionID,
-                    'LastCommentID' => null,
-                    'LastDateInserted' => val('DateInserted', $discussion)
-                ]);
-            }
-
-            $categoryModel = new CategoryModel();
-            $categoryModel->setField($categoryID, $cacheAmendment);
-            $categoryModel->setRecentPost($categoryID);
+            /** @var Vanilla\Scheduler\SchedulerInterface $scheduler */
+            $discussion = $discussion ?: null;
```

Comment:
The function is documented as only working with an array, but it looks like it would've also worked with an object. To be safe, you'll probably want to ensure it's an array here (casting objects to an array).